File: challenges/anagrams2/python/anagram2.py
def is_character_match(string1, string2):  # Performs anagram test

    # get raw character list with no whitespace to compare against each other
    string1Split = [char for char in string1.lower() if char != ' ']
    string2Split = [char for char in string2.lower() if char != ' ']

    # first, check len of strings, if false then they are not anagrams
    if len(string1Split) == len(string2Split):
        # checks characters in strings
        for char in string1Split:
            if char not in string2Split:
                return False  # false when different char is found
            else:
                return True

    # returns if strings are not equal size
    return False


def anagrams_for(word, list_of_words):
    # uses logic from anagrams1 challenge to perform anagram test
    anagrams = list(
        filter(lambda ana: is_character_match(word, ana), list_of_words))
    return anagrams

File: challenges/anagrams2/js/anagram2.js
const characterMatch = (string1, string2) => {
    // get raw characters with no whitespace
    const string1list = string1.split('').filter((char) => {
        return char !== ' '
    })
    const string2list = string2.split('').filter((char) => {
        return char !== ' '
    })

    // first, check len of strings
    if (string1list.length === string2list.length) {
        // checks characters in strings
        for (let i = 0; i < string1list.length; i++) {
            for (let j = 0; j < string2list.length; j++) {
                if (!string2list.includes(string1list[i])) {
                    return false // false when different char is found
                }
            }
        }
        return true
    }
    return false // returns if strings are not equal size
}


const anagramsFor = (word, listOfWords) => {
    // uses logic from anagrams1 challenge to perform anagram test
    anagrams = listOfWords.filter(ana => characterMatch(word, ana))
    return anagrams
}

module.exports = { anagramsFor }